src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java

```diff
@@ -1,34 +1,59 @@
 package be.seeseemelk.mockbukkit.scheduler;
 
-import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.Callable;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 import java.util.function.Consumer;
 import java.util.logging.Logger;
 
+import be.seeseemelk.mockbukkit.MockBukkit;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
 import org.bukkit.plugin.Plugin;
 import org.bukkit.scheduler.BukkitRunnable;
 import org.bukkit.scheduler.BukkitScheduler;
 import org.bukkit.scheduler.BukkitTask;
 import org.bukkit.scheduler.BukkitWorker;
 
 import be.seeseemelk.mockbukkit.UnimplementedOperationException;
+import org.jetbrains.annotations.NotNull;
 
 public class BukkitSchedulerMock implements BukkitScheduler
 {
 	private static final String LOGGER_NAME = "BukkitSchedulerMock";
+	private final ThreadPoolExecutor pool = new ThreadPoolExecutor(0, Integer.MAX_VALUE,
+			60L, TimeUnit.SECONDS,
+			new SynchronousQueue<>());
+	private final ExecutorService chatExecutor = Executors.newCachedThreadPool();
+	private final TaskList scheduledTasks = new TaskList();
+	private final AtomicReference<Exception> asyncException = new AtomicReference<>();
 	private long currentTick = 0;
 	private int id = 0;
-	private List<ScheduledTask> tasks = new LinkedList<>();
-	private ExecutorService pool = Executors.newCachedThreadPool();
-	private AtomicInteger asyncTasksRunning = new AtomicInteger();
-	private AtomicReference<Exception> asyncException = new AtomicReference<>();
-	private int asyncTasksQueued = 0;
+	private long holdExecutor = 60000;
+
+	private static Runnable runTask(ScheduledTask task) {
+		return () -> {
+			task.setRunning(true);
+			task.run();
+			task.setRunning(false);
+		};
+	}
```

Comment:
```suggestion
	private static Runnable wrapTask(ScheduledTask task) {
		return () -> {
			task.setRunning(true);
			task.run();
			task.setRunning(false);
		};
	}
```
I would consider `wrap` instead of `run` as it does not actually run the task

Reply:
^ I agree